rfc-174-consolidate-frontend-rendering-apps.md

```diff
@@ -0,0 +1,330 @@
+* Optimise the frontend code
+    * remove duplication
+* Remove the routes from the candidate
+* [Retire the candidate] once all routes have been removed
+
+### Expected issues
+Merging all of our apps together is not without difficulties and we acknowledge that a single app will present problems that our multiple apps do not.
+
+* A Rails update for one large app might be more complex than spread across individual apps.
+* The scaling of a single app might become more costly.
+* It won’t be possible to scale targeted parts of the stack (i.e. one of the existing apps) for expected increases in traffic, instead the whole site will need to be scaled.
+* It’s only possible to test one branch of code in a test environment at once. If there’s only one repo, only one change can be tested at a time.
+* It will be harder to find the pull request history for the retired repositories.
+
+In addition, we will need to address the following issues:
+
+* The test suite will take longer due to more tests.
+    * could potentially parallelise in CI
+    * will need to consider approach for developer machines
+* Need to reconfigure Kubernetes deployment. We should be able to drop the total number of pods.
+    * optimise the number of pods needed in draft and production
+    * currently in production there are
+        * 4 pods for `frontend`
+        * 6 for `government-frontend`
+        * 4 for `collections`
+        * 9 for `finder-frontend`
+        * plus 2 or 3 per app for the others
+    * scaling up the pods for a single app will use more resources
+* Resolve the question around environment testing
+    * rather than testing on integration, we should still be able to test multiple branches using preview apps
+    * would need to get new search working in the preview apps
+
+### After merging
+Once a candidate has been merged, the following steps would be needed.
+
+* Recalibrate Kubernetes deployment
+* Move components from govuk_publishing_components into the app if only being used in the frontend
+    * this information can already be found in our component auditing tools
+    * need to consider which ones are used by publishing applications
+* Begin work to remove static, slimmer and router (to be considered separately from this RFC)
+* Update component auditing
+* Decide how to handle app ownership and maintenance and what that means now we have fewer or only one app
+* Consider renaming `frontend`
+
+## Other options to explore
+
+We write this knowing that new technologies are constantly emerging that may prove useful in this area. One such technology is [Rails Engines], which allows you to wrap a specific Rails application or subset of functionality and share it with other applications or within a larger packaged application.
+
+As a separate piece of related work, we propose exploring Rails Engines as follows.
+
+### Option 1: Engines in existing app
```

Comment:
There's a conceptual difference between a monolithic app and a monolithic repo - for instance we're already committed to a situation where some updates are done in gems, some in apps (this isn't a perfect analogy, of course, because generally we're hoping to share the gem code). It comes down a bit to which you think is more serious : the architectural problems caused by multiple apps or the developer-mindspace problems caused by multiple apps. It's worth bearing in mind too that this app (unlike, say, Whitehall), is likely to end up with multiple teams working on it, so separate repos in engines do have a slight cleanness advantage over one bit repo. 

I don't want to imply I'm married to the idea of engines, though. We could just remove the options section, or reframe it as "options we have considered and rejected".